prov/shm/src/smr.h

```diff
@@ -115,6 +115,7 @@ struct smr_tx_entry {
 	void		*context;
 	struct iovec	iov[SMR_IOV_LIMIT];
 	uint32_t	iov_count;
+	void		*map_ptr;
```

Comment:
Shouldn't this be needed in the msg protocol as well? If so, it should probably be in the first patch. I also don't see any processing of the resp entry and cleaning up of the shm file in the previous commit. It should probably be there too.
We need to make sure we properly increment the cmd_cnt as well. I've added a fix and clarification for this in #5682 (adding a general progress_resp_entry function that always increments the cmd_cnt)

Reply:
In msg protocol, we don't need this map_ptr, as the sender does munmap/close as soon as it finishes copying the data into shm file. Upon receiving the signal from the receiver (resp->status==0), it just needs to shm_unlink and clean up map_name as follows.
```/c
if (pending->map_name) {
        shm_unlink(pending->map_name->name);
        dlist_remove(&pending->map_name->entry);
        free(pending->map_name);
}
```
map_ptr here is only for RMA read operation, so that the initiator does not need to get shm file name again and mmap again when progressing response.

> We need to make sure we properly increment the cmd_cnt as well. I've added a fix and clarification for this in #5682 (adding a general progress_resp_entry function that always increments the cmd_cnt)

Please see my reply in your comment `Again, see comment about looking at...`.